### PR TITLE
WIP: build managed artifacts for all *nix tests on one platform

### PR DESCRIPTION
**Pull request by trylek**
As my previous attempt at this PR shows CI errors I have a hard
time to troubleshoot, I'm reconstructing the PR incrementally
in a new branch and using CI validation to identify the bug.

Thanks

Tomas

### CODE REVIEW

**Comment by trylek** [comment]
@jkoritzinsky - I don't yet have this fully working so I'm not asking for a full review but I'd be grateful if you could take a first look at the build-test script change I had to make, otherwise managed test component build complained about missing native projects (the Error clause at line 163 in dir.targets). I have also slightly cleaned up the script so that it shows the actual MSBuild command line rather than synthesizing it twice after I spent half an hour tearing my hair out why my change to the command-line isn't kicking in before realizing I have only modified the echo :-).

**Comment by jkoritzinsky** [comment]
The change to build-test.cmd looks good to me. Just make sure to make the matching change in build-test.sh.

File: eng/run-test-job.yml [hunk 1]
```diff
@@ -69,13 +86,45 @@ jobs:
...
+        artifactName: 'NativeTestComponents-$(osIdentifier)-$(archType)-$(buildConfig)'
+        displayName: 'native test artifacts'
+
+
+    # Copy native components to the test output folder
+    - ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
+      - script: ./build-test.sh copynativeonly skipgeneratelayout $(buildConfig) $(archType) $(priorityArg) ci
+        displayName: Copy native components to test output folder
+    - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+      - script: build-test.cmd skipnative skipgeneratelayout $(buildConfig) $(archType) $(priorityArg) ci
```
Comment (jkoritzinsky) [inline eng/run-test-job.yml]:
```suggestion
      - script: build-test.cmd copynativeonly skipgeneratelayout $(buildConfig) $(archType) $(priorityArg) ci
```

Comment (trylek) [reply eng/run-test-job.yml]: Ahh, damn copy-paste, thanks for spotting!

**Comment by trylek** [comment]
@jkoritzinsky - Could you please take a look at the latest batch of test executions? I suspect there might be some problem related to test priority in the copynativeonly option. It seems to me that the command is copying Pri#1 tests although we're not passing the -priority=1 to the build-test script.

**Comment by trylek** [comment]
@jkoritzinsky - Hmm, I'm starting to worry that running the copynativeonly step during the run job is a big bug farm. What are the concerns against just copying the native bits to the test output folders during product build, zipping them from there and just unpacking them into the same location in the run step?

**Comment by jkoritzinsky** [comment]
> What are the concerns against just copying the native bits to the test output folders during product build, zipping them from there and just unpacking them into the same location in the run step?

As long as we use the copynativeonly to place them into their final locations (instead of just overlaying the obj directory on top of the bin directory), then that should work.

**Comment by trylek** [comment]
Closing this WIP review as I'm trying to pursue a different fix for the issue.

**Pull request closed without merging**